src/main/java/nc/students/ayaz/model/User.java

```diff
@@ -1,46 +1,62 @@
 package nc.students.ayaz.model;
 
 import cobertura.IgnoreDuringCodeCoverage;
-import nc.students.ayaz.model.exceptions.NoSuchVideoException;
+import nc.students.ayaz.model.exceptions.NoSuchRecourceException;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
 
 public class User {
 
     private String nickname;
 
-    private List<Video> videos;
+    private Map<String, Video> videos;
 
     public User(String nickname) {
         this.nickname = nickname;
-        videos = new ArrayList<>();
+        this.videos = new HashMap<>();
     }
 
     public String getNickname() {
         return nickname;
     }
 
-    public List<Video> getVideos() {
+    public Map<String, Video> getVideos() {
         return videos;
     }
 
     public void addVideo(Video video) {
-        videos.add(video);
+        videos.put(video.getName(), video);
     }
 
     public boolean ownsVideo(Video video) {
```

Comment:
&#x1F34F; **Resolved**
Agree. Sorry, i forgot to change when i changed from List to Map.